### PR TITLE
SDIT-1315: ✨ Add in secured endpoint

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/ResourceSecurityTest.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          mappingInfo.getMappings().forEach {
+            assertThat(exclusions.contains(it)).withFailMessage {
+              "Found $mappingInfo of type $method with no PreAuthorize annotation"
+            }.isTrue()
+          }
+        }
+      }
+    }
+  }
+}
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/ResourceSecurityTest.kt:51]: Maybe put the contents of this into a bean in the test library - then this test could inject the bean and call the lib.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/wiremock/HmppsAuthMockServer.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import com.github.tomakehurst.wiremock.client.WireMock.post
+import com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo
+import com.github.tomakehurst.wiremock.http.HttpHeader
+import com.github.tomakehurst.wiremock.http.HttpHeaders
+import org.junit.jupiter.api.extension.AfterAllCallback
+import org.junit.jupiter.api.extension.BeforeAllCallback
+import org.junit.jupiter.api.extension.BeforeEachCallback
+import org.junit.jupiter.api.extension.ExtensionContext
+
+class HmppsAuthApiExtension : BeforeAllCallback, AfterAllCallback, BeforeEachCallback {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/wiremock/HmppsAuthMockServer.kt:15]: I wonder if this extension should go in the test library?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/config/HmppsTemplateKotlinExceptionHandler.kt [hunk 3]
```diff
@@ -35,6 +36,17 @@ class HmppsTemplateKotlinExceptionHandler {
       ),
     ).also { log.info("No resource found exception: {}", e.message) }
 
+  @ExceptionHandler(AccessDeniedException::class)
+  fun handleAccessDeniedException(e: AccessDeniedException): ResponseEntity<uk.gov.justice.hmpps.kotlin.common.ErrorResponse> = ResponseEntity
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/config/HmppsTemplateKotlinExceptionHandler.kt:40]: Not related to this PR, just something to ponder - would we ever add an exception handler to the library? I think the ErrorResponse is already in there.

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/config/HmppsTemplateKotlinExceptionHandler.kt (hunk 3 above)
Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/config/HmppsTemplateKotlinExceptionHandler.kt:40]: I _think_ we could add some default advice with
```
@Order(Ordered.LOWEST_PRECEDENCE)
``` that would then be overridden by any other advice.  Therefore sounds a simple case of just adding the handler to the library.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/wiremock/HmppsAuthMockServer.kt (hunk 2 above)
Comment (petergphillips) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/wiremock/HmppsAuthMockServer.kt:15]: yes, sounds reasonable 👍

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/ResourceSecurityTest.kt (hunk 1 above)
Comment (petergphillips) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/ResourceSecurityTest.kt:51]: agreed

**Pull request merged**